### PR TITLE
Answer chapter 1

### PR DESCRIPTION
**Pull request by thilinatnt**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -619,9 +631,10 @@ Implement a function that returns the sum of the last two digits of a number.
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
-
+sumLast2 :: Int -> Int
+sumLast2 n = 
+    let last2Dig = mod (abs n) 100
+    in mod last2Dig 10 + div last2Dig 10
```
Comment (vrom911) [inline src/Chapter1.hs:637]: That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: the standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(lastDiv, lastMod) = divMod last2Dig 10
```

You can see how we could pattern match on the pair 🙂 

And use it in the sum 👍🏼

File: src/Chapter1.hs [hunk 2]
```diff
@@ -641,7 +654,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit n = if abs n < 10 then abs n else divisible n
```
Comment (vrom911) [inline src/Chapter1.hs:657]: You can pass `abs n` to `divisible` so you won't need to use `abs` inside 🙂 

You can see that `abs n` is reused several times in this function already. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼

**Review by vrom911 — APPROVED** [review]
Great start! Congrats with nailing Chapter One 🎉 
Feel free to merge this PR as it already looks wonderful 👍🏼

File: src/Chapter1.hs (hunk 1 above)
Comment (thilinatnt) [reply src/Chapter1.hs:637]: Wow that's a really nice function
Thanks! for the heads up 👍

File: src/Chapter1.hs (hunk 2 above)
Comment (thilinatnt) [reply src/Chapter1.hs:657]: Thanks! for the comment 👍

**Pull request merged**